### PR TITLE
add third party signin button to hamburger menu

### LINKED ISSUE #6937: Add Google sign in button to hamburger menu
One of the nice things about the google sign in button is that it requires only a single click. Let's add this button to the hamburger menu, so patrons can log in with a single click from anywhere.

### Proposal & Constraints

| Currently | After |
| ---- | --- |
| image | image |

Constraints:

- The iframe must not load until the hamburger icon is clicked
- The iframe should only load for logged out users

### Stakeholders
@mekarpeles

### PR DESCRIPTION
**Pull request by JaydenTeoh**
Closes #6937 

Added third party login button under hamburger menu

### Screenshot

My Open Library

### Stakeholders

@cdrini created issue

### CODE REVIEW

**Comment by mekarpeles** [comment]
@cdrini my concern is around any sort of iframe tracking across the website. Do you mind investigating?

**Comment by cdrini** [comment]
We'll need someway to show/hide the iframe depending on if the hamburger menu has been opened. We want to keep these pieces relatively decoupled, so don't want the iframe eg listening to an event. Instead would prefer the parent component, the header dropdown, to call the iframe when it sees fit.

We can do this by having some global function inside the ia_third_party_logins iframe to act as an API, and then from the header_dropdown, we can listen to the [toggle event](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#events) on the details element, and call the methods as needed

**Comment by JaydenTeoh** [comment]
@cdrini @mekarpeles I have implemented lazy loading of the iframe using window events but I am unable to unload/change the `src` attribute of the iframe to an empty string upon closing the hamburger menu. I believe this is due to the violation of the frame-ancestors directive of the implemented Content Security Policy (CSP). Is there any way around this?

File: openlibrary/templates/account/ia_thirdparty_logins.html [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+<script>
+    window.addEventListener('LOAD_THIRD_PARTY_LOGINS', function() {
+        var iframe = document.getElementById('ia-third-party-logins');
+        iframe.src = iframe.getAttribute('data-src')
+    });
+
+    window.addEventListener('UNLOAD_THIRD_PARTY_LOGINS', function() {
+        var iframe = document.getElementById('ia-third-party-logins');
+        iframe.src = '';
+    });
```
Comment (cdrini) [inline openlibrary/templates/account/ia_thirdparty_logins.html:20]: Let's define these as globals:
```suggestion
    window.LOAD_THIRD_PARTY_LOGINS = function() {
        var iframe = document.getElementById('ia-third-party-logins');
        iframe.src = iframe.getAttribute('data-src');
    };

    window.UNLOAD_THIRD_PARTY_LOGINS = function() {
        var iframe = document.getElementById('ia-third-party-logins');
        iframe.src = '';
    };

```

Then call these functions directly with eg `window.LOAD_THIRD_PARTY_LOGINS()`. The event handling creates an extra level of misdireciton, and uses some things which would require polyfilling (eg `Event`)

**Comment by JaydenTeoh** [comment]
@cdrini I have made the changes requested and now the `UNLOAD_THIRD_PARTY_LOGINS` function is also successful in changing the HTML src attribute of the iframe to an empty string! Not sure why it works now after changing to global functions even though I still get the CSP warning in the console.

Anyways, here are the changes I made according to your comments:

1. Change loading and unloading into global window functions
2. Added a new class to the details element in `header_dropdown.html` (`<details class="$(props['name'])-details">`)
3. Added the following watch function in `openlibrary/js/index.js`:
```
// Load and unload third party login iframes when toggling hamburger menu
    const hamburgerMenu = document.querySelector('.hamburger-details')
    hamburgerMenu.addEventListener('toggle', () => {
        if (hamburgerMenu.open) {
            window.LOAD_THIRD_PARTY_LOGINS()
        } else {
            window.UNLOAD_THIRD_PARTY_LOGINS()
        }
    })
```

Do let me know if I missed out on anything!

**Review by cdrini — APPROVED** [review]
Niiice! Working smoothly :) A few small changes. Also pushing some change to archive.org side to make the iframe a little nicer:

image

**Comment by cdrini** [comment]
This looks good! Will merge once the archive.org changes are merged through :+1:

**Comment by cdrini** [comment]
Deployed on IA! :+1:

**Pull request merged**